consensus/spos/worker.go

```diff
@@ -325,6 +325,13 @@ func (wrk *Worker) ProcessReceivedMessage(message p2p.MessageP2P, fromConnectedP
 	cnsMsg := &consensus.Message{}
 	err = wrk.marshalizer.Unmarshal(cnsMsg, message.Data())
 	if err != nil {
+		//this situation is so severe that we have to black list both the message originator and the connected peer
+		//that disseminated this message.
+
+		reason := "blacklisted due to invalid consensus message"
+		wrk.antifloodHandler.BlacklistPeer(message.Peer(), reason, core.IntervalBlackListPeerInvalidMessage)
+		wrk.antifloodHandler.BlacklistPeer(fromConnectedPeer, reason, core.IntervalBlackListPeerInvalidMessage)
```

Comment:
as discussed, please make sure a peer that sends you the message but it's not the message's originator won't get blacklisted if it did nothing wrong.

Reply:
Good catch, implemented unjoin, pending the integration test fixing

Reply:
Done testing, removed one antiflood bug because of start in epoch mechanism